packages/@aws-cdk-testing/framework-integ/test/aws-appsync/test/integ.appsync-function-max-batch-size.ts

```diff
@@ -0,0 +1,33 @@
+import * as path from 'path';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as cdk from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as appsync from 'aws-cdk-lib/aws-appsync';
+import { STANDARD_NODEJS_RUNTIME } from '../../config';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-appsync-function-max-batch-size');
+
+const api = new appsync.GraphqlApi(stack, 'LambdaAPI', {
+  name: 'LambdaAPI',
+  schema: appsync.SchemaFile.fromAsset(path.join(__dirname, 'appsync.lambda.graphql')),
+});
+
+const func = new lambda.Function(stack, 'func', {
+  code: lambda.Code.fromAsset(path.join(__dirname, 'verify', 'lambda-tutorial')),
+  handler: 'lambda-tutorial.handler',
+  runtime: STANDARD_NODEJS_RUNTIME,
+});
+
+const lambdaDS = api.addLambdaDataSource('LambdaDS', func);
+
+lambdaDS.createFunction('TestFunction', {
+  name: 'test',
+  maxBatchSize: 10,
+});
+
```

Comment:
I assume this is the same as doing `api.AppsyncFunction(this, ...`?

Reply:
Yes, this creates `api.AppsyncFunction`.

https://github.com/aws/aws-cdk/blob/511495506abec7864cc0ba16608e92b8fd5621c1/packages/aws-cdk-lib/aws-appsync/lib/data-source.ts#L155